src/ariel-os-hal/src/define_peripherals.rs

```diff
@@ -61,7 +61,7 @@ macro_rules! group_peripherals {
         $group:ident {
             $(
                 $(#[$inner:meta])*
-                $peripheral_name:ident : $peripherals:ident
+                $peripheral_name:ident : $peripherals:ty
```

Comment:
Looking at [the definition](https://doc.rust-lang.org/reference/types.html#grammar-Type), `ty` seems maybe a bit too generic, maybe that should be `path` (a type path) instead:

```suggestion
                $peripheral_name:ident : $peripherals:path
```

Reply:
works too, but a path is actually more generic, as in, it resolves to Type at some point: https://doc.rust-lang.org/reference/paths.html#r-paths.type

Reply:
> a path is actually more generic

I don't think it is, `Type` only appears *within* `TypePathFn`, while `TypePath` is one possible `Type`.